source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -463,18 +519,65 @@ SplitRequestPtr InstanceImpl::makeRequest(Common::Redis::RespValuePtr&& request,
     return nullptr;
   }
 
+  // Get the handler for the downstream request
   auto handler = handler_lookup_table_.find(to_lower_string.c_str());
   if (handler == nullptr) {
     stats_.unsupported_command_.inc();
     callbacks.onResponse(Common::Redis::Utility::makeError(
         fmt::format("unsupported command '{}'", request->asArray()[0].asString())));
     return nullptr;
   }
+
+  // Fault Injection Check
+  absl::optional<std::pair<Common::Redis::FaultType, std::chrono::milliseconds>> fault =
+      fault_manager_->getFaultForCommand(to_lower_string);
+
+  // Start request
   ENVOY_LOG(debug, "redis: splitting '{}'", request->toString());
   handler->command_stats_.total_.inc();
-  SplitRequestPtr request_ptr = handler->handler_.get().startRequest(
-      std::move(request), callbacks, handler->command_stats_, time_source_);
-  return request_ptr;
+  if (!fault.has_value()) {
```

Comment:
The logic here seems to be quite convoluted.  Can we simplify it a little please?  something alone the line of:
if (fault.has_value() && fault type == Error)
  request_ptr = ... error fault
else
 request_ptr = ... normal request
if (fault.has_value and delay > 0)
  ... wrap request_ptr in delay fault
  return delay_fault_ptr
else
  return request_ptr